syntax = "proto3";

package mantis.persona.v1;

// Persona characteristics and behavioral patterns extension
// URI: https://mantis.ai/extensions/persona-characteristics/v1
message PersonaCharacteristics {
  // Core principles that guide the persona's worldview and decisions
  repeated string core_principles = 1;
  
  // Framework for how the persona approaches decision-making
  string decision_framework = 2;
  
  // Style and tone of communication preferred by the persona
  string communication_style = 3;
  
  // Patterns of thought and cognitive approaches
  repeated string thinking_patterns = 4;
  
  // Characteristic phrases and expressions the persona uses
  repeated string characteristic_phrases = 5;
  
  // Behavioral tendencies and habits
  repeated string behavioral_tendencies = 6;
  
  // Original content used to generate the persona. Can be used as a system prompt.
  string original_content = 7;
}

// Role adaptation scores and preferences
message RoleAdaptation {
  // Score for effectiveness as a leader (0.0 - 1.0)
  float leader_score = 1;
  
  // Score for effectiveness as a follower (0.0 - 1.0)
  float follower_score = 2;
  
  // Score for effectiveness as a narrator/observer (0.0 - 1.0)
  float narrator_score = 3;
  
  // Preferred role in team dynamics
  string preferred_role = 4;
  
  // Flexibility in adapting to different roles (0.0 - 1.0)
  float role_flexibility = 5;
}

// Competency scores across key skill areas
// URI: https://mantis.ai/extensions/competency-scores/v1
message CompetencyScores {
  // Map of competency name to score (0.0 - 1.0)
  // Common competencies include:
  // - "strategic planning and long-term vision"
  // - "team leadership and inspiring others"
  // - "decisive decision making under pressure"
  // - "clear and persuasive communication"
  // - "analytical thinking and logical reasoning"
  // - "creative innovation and design thinking"
  // - "risk assessment and mitigation planning"
  // - "stakeholder relationship management"
  // - "domain expertise and technical knowledge"
  // - "adaptability to changing circumstances"
  map<string, float> competency_scores = 1;
  
  // Role adaptation scores and preferences
  RoleAdaptation role_adaptation = 2;
}

// Domain expertise and knowledge areas
// URI: https://mantis.ai/extensions/domain-expertise/v1
message DomainExpertise {
  // Primary domains of expertise
  repeated string primary_domains = 1;
  
  // Secondary areas of knowledge
  repeated string secondary_domains = 2;
  
  // Methodologies and approaches the persona employs
  repeated string methodologies = 3;
  
  // Tools and frameworks the persona is familiar with
  repeated string tools_and_frameworks = 4;
}